### PR TITLE
MRG: Add Evoked.decimate

### LINKED ISSUE #3287: ENH: evoked.decim
It would be nice to be able to decimate evoked data like we can with `Epochs`. It should be fairly straightforward.

### LINKED ISSUE #3296: BUG: problems with epochs._raw_times while cropping resampled epochs
While cropping epochs, we also change epochs._raw_times. However, as soon as epochs were resampled before cropping, epochs._raw_times is based on a different sfreq. Thats why we cannot apply the same tmask for epochs.times and epochs._raw_times. This leeds to wrong values in epochs._raw_times if sfreq_new < sfreq_old and an error if sfreq_new > sfreq_old. Changing epochs.crop(). We thus should use a different tmask for changing epochs._raw_times. As we do not store the old sampling frequency (AFAIK), we need to compute it as 1./ (epochs._raw_times[1]-epochs._raw_times[0]).

In mne.epochs.crop(), changing
```
tmask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
self.times = self.times[tmask]
self._raw_times = self._raw_times[tmask]
self._data = self._data[:, :, tmask]
```
to 
```
tmask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
self.times = self.times[tmask]
self._data = self._data[:, :, tmask]
tmask = _time_mask(self._raw_times, tmin, tmax,
    sfreq=(1. / (self._raw_times[1] - self._raw_times[0])))
self._raw_times = self._raw_times[tmask]
```
should solve the issue.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Also fixes a bug with existing `Epochs.decimate` when the `offset` parameter is used.

Closes #3287.
Closes #3296.

Ready for review/merge from my end.

### CODE REVIEW

File: mne/evoked.py [hunk 1]
```diff
@@ -160,6 +160,44 @@ def crop(self, tmin=None, tmax=None):
...
+            current sampling rate.
+
+        Returns
+        -------
+        evoked : instance of Evoked
+            The decimated Evoked object.
+
+        See Also
+        --------
+        Epochs.decimate
```
Comment (agramfort) [inline mne/evoked.py]: resample?

File: mne/tests/test_evoked.py [hunk 2]
```diff
@@ -16,20 +16,61 @@
...
+    epochs = EpochsArray(data, info, events)
+    data_epochs = epochs.copy().decimate(decim).get_data()
+    data_epochs_2 = epochs.copy().decimate(decim, offset=1).get_data()
+    data_epochs_3 = epochs.decimate(dec_1).decimate(dec_2).get_data()
+    assert_array_equal(data_epochs, data[:, :, ::decim])
+    assert_array_equal(data_epochs_2, data[:, :, 1::decim])
+    assert_array_equal(data_epochs, data_epochs_3)
+
+    # Now let's do it with some real data
+    raw = io.Raw(raw_fname)
```
Comment (agramfort) [inline mne/tests/test_evoked.py]: read_raw_fif

**Comment by agramfort** [comment]
you need to rebase

besides LGTM

**Comment by jaeilepp** [comment]
LGTM. Just remember to add the update of ``_raw_times`` to ``epochs.resample``.

**Comment by Eric89GXL** [comment]
Comments addressed

**Comment by jaeilepp** [comment]
LGTM

**Comment by Eric89GXL** [comment]
Thanks for the reviews

**Pull request merged**